test/e2e/azure_clusterproxy.go

```diff
@@ -203,6 +208,57 @@ func (acp *AzureClusterProxy) collectPodLogs(ctx context.Context, namespace stri
 	}
 }
 
+func (acp *AzureClusterProxy) collectNodes(ctx context.Context, namespace string, name string, aboveMachinesPath string) {
+	workload := acp.GetWorkloadCluster(ctx, namespace, name)
+	nodes := &corev1.NodeList{}
+
+	Expect(workload.GetClient().List(ctx, nodes)).To(Succeed())
+
+	var err error
+	var nodeDescribe string
+
+	nodeDescriber, ok := describe.DescriberFor(schema.GroupKind{Group: corev1.GroupName, Kind: "Node"}, workload.GetRESTConfig())
+	if !ok {
+		Logf("failed to get node describer")
+	}
+
+	for _, node := range nodes.Items {
+		// Describe the node.
+		Logf("Describing Node %s", node.GetName())
+		nodeDescribe, err = nodeDescriber.Describe(node.GetNamespace(), node.GetName(), describe.DescriberSettings{ShowEvents: true})
+		if err != nil {
+			Logf("failed to describe node %s: %v", node.GetName(), err)
+		}
+
+		go func(node corev1.Node) {
+			defer GinkgoRecover()
+
+			describeFile := path.Join(aboveMachinesPath, nodesDir, node.GetName(), "node-describe.txt")
+			if err := os.MkdirAll(filepath.Dir(describeFile), 0o755); err != nil {
+				// Failing to mkdir should not cause the test to fail
+				Logf("Error mkdir: %v", err)
+				return
+			}
+
+			f, err := os.OpenFile(describeFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+			if err != nil {
+				// Failing to open the file should not cause the test to fail
+				Logf("Error opening file to write node describe: %v", err)
+				return
+			}
+			defer f.Close()
+
+			out := bufio.NewWriter(f)
+			defer out.Flush()
+			_, err = out.WriteString(nodeDescribe)
+			if errors.Is(err, io.ErrUnexpectedEOF) {
```

Comment:
Do we need to check that it's an unexpected EOF error here, or should any possible error from `WriteString` not cause the test to fail?

Reply:
if it's EOF we don't even log the failure, if it's any other error we log the failure but don't fail the test

Reply:
Makes sense. Should this be inverted then?
```go
if !errors.Is(err, io.ErrUnexpectedEOF) {
```

Reply:
now that I'm re-reading this, I think the intent is to log the failure if the describe string is empty, but we never want to return an error. Although not sure if that makes sense since we're already logging an error above if `Describe()` returns an error https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3179/files/a25d333551730fed0f1a32a46f09ed89ac39bf69#diff-34d5acd96ae3a8b3040a0d8a711797ef5eaceabe05de690d663fa6cb670cead5R230

To be honest i took this from https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3179/files/a25d333551730fed0f1a32a46f09ed89ac39bf69#diff-34d5acd96ae3a8b3040a0d8a711797ef5eaceabe05de690d663fa6cb670cead5L198 without really questioning it, let me see if I can refactor it to reuse the same logic and look more closely to see if it's doing what we expect.

Reply:
did a refactor, PTAL

Reply:
I did, looks good. 👍🏻 Just want to see the describe artifacts it produces.